res/controllers/midi-components-0.0.js

```diff
@@ -158,6 +162,26 @@
                 }
             }
         },
+        /**
+         * @typedef {object} ControlDefinition
+         * @property {number} status MIDI status
+         * @property {number} midino MIDI number
+         * eslint-disable-next-line jsdoc/no-undefined-types
+         * @property {MidiInputHandler?} control MIDI input handler
```

Comment:
I don't know about the naming here. Especially about `control` for the callback. I wanted to reuse the wording used in the XML but it bears the legacy from when XML was used to plug MIDI signals to Mixxx controls.

Reply:
well ideally, the design was just that the data passed to `midi:` was automatically used to register `input`. The problem is that this is a breaking API change (since that would also now register mappings that have manual XML mappings). IMO the ethos of ComponentJS is "one IO per component", so having multiple input handlers and multiple outputhandlers backed into the base component doesn't make sense.

Reply:
I see. The only problem I have with using `midi`, is that it's already used to connect Mixxx controls with device LED in `output`.

Reply:
yeah, we need to ensure the old behavior doesn't change, but we can introduce a new property to opt into the new behavior.